app/scripts/modules/core/src/search/global/GlobalSearch.tsx

```diff
@@ -173,92 +214,120 @@ export class GlobalSearch extends React.Component<{}, IGlobalSearchState> {
 
   @Debounce(300)
   private executeQuery() {
```

Comment:
I'd probs move this into the rxjs chain via [debounceTime](https://www.learnrxjs.io/operators/filtering/debouncetime.html)

```
this.query$
      .debounceTime(300)
      .distinctUntilChanged()
      .do(() => {
        ReactGA.event({ category: 'Global Search', action: 'Query', label: query });
        this.setState({ querying: true });
      })
      .switchMap((query: string) => (
        Observable.fromPromise(search.query(query))
      ))
...   
```

Reply:
i was originally considering moving the debounce, but the previous observable library i used very strongly discouraged side effects outside of an explicit subscription — sounds like we're not averse to some thoughtfully-placed `do`s?

Reply:
i ended up doing this, _but_ i left off the `distinctUntilChanged()` because this Subject is only getting values pushed into it when certain conditions make searching necessary (i.e. query length). Because of that this would stall and do nothing if you typed in `deck`, deleted the text (making it too short to trigger a search), and then typed in `deck` again (since all Subject saw was `deck` and then `deck` again). The logic that decides when to push into this Subject is part of an `if/else if` so I can't really move the logic into the observable chain itself 🤷‍♂️ .